beacon_node/http_api/src/ui.rs

```diff
@@ -0,0 +1,175 @@
+use lighthouse_network::{types::SyncState, NetworkGlobals};
+use parking_lot::RwLock;
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+use std::sync::Arc;
+use sysinfo::{CpuExt, DiskExt, NetworkExt, NetworksExt, System, SystemExt};
+use types::EthSpec;
+
+/// System related health, specific to the UI.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct SystemHealth {
+    /// Total memory of the system.
+    pub total_memory: u64,
+    /// Total free memory available to the system.
+    pub free_memory: u64,
+    /// Total used memory.
+    pub used_memory: u64,
+
+    /// System load average over 1 minute.
+    pub sys_loadavg_1: f64,
+    /// System load average over 5 minutes.
+    pub sys_loadavg_5: f64,
+    /// System load average over 15 minutes.
+    pub sys_loadavg_15: f64,
+
+    /// Total cpu cores.
+    pub cpu_cores: usize,
+    /// Total cpu threads.
+    pub cpu_threads: usize,
+    /// The global cpu frequency.
+    pub global_cpu_frequency: f32,
+
+    /// Total capacity of disk.
+    pub disk_bytes_total: u64,
+    /// Free space in disk.
+    pub disk_bytes_free: u64,
+
+    /// The name of the network that uses the most traffic.
+    pub network_name: String,
+    /// Total bytes received over the main interface.
+    pub network_bytes_total_received: u64,
+    /// Total bytes sent over the main interface.
+    pub network_bytes_total_transmit: u64,
+
+    /// The current NAT status.
+    pub nat_open: bool,
+    /// The current number of connected peers.
+    pub connected_peers: usize,
+    /// The current syncing state of the consensus node.
+    pub sync_state: SyncState,
+
+    /// System uptime.
+    pub system_uptime: u64,
+    /// Application uptime.
+    pub app_uptime: u64,
+    /// The System name
+    pub system_name: String,
+    /// Kernel version
+    pub kernel_version: String,
+    /// OS version
+    pub os_version: String,
+    /// Hostname
+    pub host_name: String,
+}
+
+impl SystemHealth {
+    /// Populates the system health.
+    pub fn observe<TSpec: EthSpec>(
+        sysinfo: Arc<RwLock<System>>,
+        app_uptime: u64,
+        network_globals: Arc<NetworkGlobals<TSpec>>,
+    ) -> Self {
+        let sysinfo = sysinfo.read();
+        let loadavg = sysinfo.load_average();
+
+        let cpus = sysinfo.cpus();
+
+        let disks = sysinfo.disks();
+
+        let networks = sysinfo.networks();
+
+        let system_uptime = sysinfo.uptime();
+
+        // Helper functions to extract specific data
+
+        // Find the root fs and report this
+        let (disk_bytes_total, disk_bytes_free) = {
+            let individual_disk = disks
+                .iter()
+                .find(|disk| {
+                    disk.mount_point() == Path::new("/")
```

Comment:
This is going to be kind of wonky if the user has their data directory on a disk other than `/`, which is probably quite a common setup.

I think it wouldn't be too hard to plumb the datadir through this function and then check which of the drives has a matching device ID (using `.dev()`). This would only work on UNIX, but that covers the majority of users.

https://stackoverflow.com/questions/69021308/determine-if-two-paths-are-on-the-same-filesystem